internal/dag/gatewayapi_processor_test.go

```diff
@@ -184,9 +193,199 @@ func TestComputeHosts(t *testing.T) {
+				},
+			},
+			want: []string(nil),
+			wantError: []error{
+				fmt.Errorf("invalid hostname \"*.*.projectcontour.io\": [a wildcard DNS-1123 subdomain must start with '*.', followed by a valid DNS subdomain, which must consist of lower case alphanumeric characters, '-' or '.' and end with an alphanumeric character (e.g. '*.example.com', regex used for validation is '\\*\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')]"),
+			},
+		},
+		"listener host & route host do not exactly match": {
+			listenerHost: listenerHostname("listener.projectcontour.io"),
+			route: &gatewayapi_v1alpha1.HTTPRoute{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "basic",
+					Namespace: "projectcontour",
+					Labels: map[string]string{
+						"app":  "contour",
+						"type": "controller",
+					},
+				},
+				Spec: gatewayapi_v1alpha1.HTTPRouteSpec{
+					Hostnames: []gatewayapi_v1alpha1.Hostname{
+						"http.projectcontour.io",
+					},
+				},
+			},
+			want:      nil,
+			wantError: []error{fmt.Errorf("gateway hostname \"listener.projectcontour.io\" does not match route hostname \"http.projectcontour.io\"")},
+		},
+		"listener host & route host exactly match": {
+			listenerHost: listenerHostname("http.projectcontour.io"),
+			route: &gatewayapi_v1alpha1.HTTPRoute{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "basic",
+					Namespace: "projectcontour",
+					Labels: map[string]string{
+						"app":  "contour",
+						"type": "controller",
+					},
+				},
+				Spec: gatewayapi_v1alpha1.HTTPRouteSpec{
+					Hostnames: []gatewayapi_v1alpha1.Hostname{
+						"http.projectcontour.io",
+					},
+				},
+			},
+			want:      []string{"http.projectcontour.io"},
+			wantError: nil,
+		},
+		"listener host & route host match wildcard host": {
+			listenerHost: listenerHostname("*.projectcontour.io"),
+			route: &gatewayapi_v1alpha1.HTTPRoute{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "basic",
+					Namespace: "projectcontour",
+					Labels: map[string]string{
+						"app":  "contour",
+						"type": "controller",
+					},
+				},
+				Spec: gatewayapi_v1alpha1.HTTPRouteSpec{
+					Hostnames: []gatewayapi_v1alpha1.Hostname{
+						"http.projectcontour.io",
+					},
+				},
+			},
+			want:      []string{"http.projectcontour.io"},
+			wantError: nil,
+		},
+		"listener host & route host do not match wildcard host": {
+			listenerHost: listenerHostname("*.projectcontour.io"),
+			route: &gatewayapi_v1alpha1.HTTPRoute{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "basic",
+					Namespace: "projectcontour",
+					Labels: map[string]string{
+						"app":  "contour",
+						"type": "controller",
+					},
+				},
+				Spec: gatewayapi_v1alpha1.HTTPRouteSpec{
+					Hostnames: []gatewayapi_v1alpha1.Hostname{
+						"http.example.com",
+					},
+				},
+			},
+			want:      nil,
+			wantError: []error{fmt.Errorf("gateway hostname \"*.projectcontour.io\" does not match route hostname \"http.example.com\"")},
+		},
```

Comment:
it'd be nice to add some cases to verify that e.g. `too.many.labels.projectcontour.io` doesn't match `*.projectcontour.io` (in both directions, i.e. where the listener has the wildcard and where the route has the wildcard) to get better coverage around the regex.

Reply:
yeah good point, probably need an integration test as well for wildcards if we're intending on supporting it already

Reply:
I added some tests, I'll make some additional integration tests in a future PR.